eras/shelley/impl/src/Cardano/Ledger/Shelley/EpochBoundary.hs

```diff
@@ -79,8 +84,10 @@ deriving newtype instance Typeable crypto => NoThunks (Stake crypto)
 deriving newtype instance
   CC.Crypto crypto => ToCBOR (Stake crypto)
 
-deriving newtype instance
-  CC.Crypto crypto => FromCBOR (Stake crypto)
+instance CC.Crypto crypto => FromSharedCBOR (Stake crypto) where
+  type Share (Stake crypto) = Share (VMap VB VP (Credential 'Staking crypto) (CompactForm Coin))
```

Comment:
Unfortunately we cannot use GND, which is caused by associated type family and it's usage in `StateT`. Beats me why ghc can't figure it out, but this is the error we get if we try to derive it:
```haskell
src/Cardano/Ledger/Shelley/EpochBoundary.hs:(87,1)-(88,51): error: …
    • Couldn't match type ‘VMap
                             VB VP (Credential 'Staking crypto) (CompactForm Coin)’
                     with ‘Stake crypto’
        arising from a use of ‘ghc-prim-0.6.1:GHC.Prim.coerce’
    • In the expression:
          ghc-prim-0.6.1:GHC.Prim.coerce
            @(Control.Monad.Trans.State.Strict.StateT (Share (VMap VB VP (Credential Staking crypto) (CompactForm Coin))) (Codec.CBOR.Decoding.Decoder s) (VMap VB VP (Credential Staking crypto) (CompactForm Coin)))
            @(Control.Monad.Trans.State.Strict.StateT (Share (Stake crypto)) (Codec.CBOR.Decoding.Decoder s) (Stake crypto))
            (fromSharedPlusCBOR
               @(VMap VB VP (Credential Staking crypto) (CompactForm Coin))) ::
            forall (s :: TYPE ghc-prim-0.6.1:GHC.Types.LiftedRep).
            Control.Monad.Trans.State.Strict.StateT (Share (Stake crypto)) (Codec.CBOR.Decoding.Decoder s) (Stake crypto)
```

Reply:
Yeah, if it wasn't for this problem this whole sharing approach could be pushed further up the stream into `FromCBOR` class and it would have been backwards compatible. But inability to derive for `newtypes` is a deal breaker on doing that, althoughusing an extra `FromSharedCBOR` is that terrible IMHO